### PR TITLE
Improve documentation for binlog code example

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
Adds a description of the example

Fixes bugs in the example code

### CODE REVIEW

**Comment by andreas-bok-sociomantic** [comment]
rebased on upstream/master

File: docs/examples/binlog.rst [hunk 1]
```diff
@@ -1,11 +1,55 @@
...
+mysql binlog.
+
+First a :c:type:`drizzle_binlog_st` is created by calling :c:func:`drizzle_binlog_init`
+with an already initialized :c:type:`drizzle_st` connection. Two callbacks are
+specified; ``binlog_event``, which is called every time a binlog event is
+retrieved and ``binlog_error`` which is called when an error occurs while parsing
+binlog-data.
+In addition it is possible to pass a user context to be used for the callback
+functions. When ``NULL`` is passed, as in this example, the context is unspecified.
+Calling :c:func:`drizzle_binlog_start` start the parsing of the binlog. The second
```
Comment (ben-palmer-sociomantic) [inline docs/examples/binlog.rst]: Should be `starts the parsing`.

**Comment by ben-palmer-sociomantic** [comment]
I had a minor comment on the documentation. The only other issue I had was that the links for the methods in the documents just seem to link back to this document.

**Comment by andreas-bok-sociomantic** [comment]
> The only other issue I had was that the links for the methods in the documents just seem to link back to this document.

Yes, the link formatting is specific the sphinx preprocessor which builds the documentation. I checked that it works in the generated documentation.

**Comment by andreas-bok-sociomantic** [comment]
Updated based on comments from @ben-palmer-sociomantic 

``` diff
diff --git a/docs/examples/binlog.rst b/docs/examples/binlog.rst
index ce81788..4155ec2 100644
--- a/docs/examples/binlog.rst
+++ b/docs/examples/binlog.rst
@@ -14,7 +14,7 @@ 
-Calling :c:func:`drizzle_binlog_start` start the parsing of the binlog. The second
+Calling :c:func:`drizzle_binlog_start` starts the parsing of the binlog. The second
```

**Comment by ben-palmer-sociomantic** [comment]
> Yes, the link formatting is specific the sphinx preprocessor which builds the documentation. I checked that it works in the generated documentation.

Ok got it, LGTM.

**Comment by ben-palmer-sociomantic** [comment]
This pull request has been rebased via `git hub pull rebase`. Original pull request HEAD was fb7af9d019b6c2052988e65c84393188e084ff2b, new (rebased) HEAD is 07d1d3e3d245d953e9c6d4b28324d3eab6368820

**Comment by andreas-bok-sociomantic** [comment]
TX!

**Pull request closed without merging**